client/my-sites/people/edit-team-member-form/edit-user-form.jsx

```diff
@@ -315,17 +321,43 @@ class EditUserForm extends Component {
 	}
 }
 
+const withExternalContributors = createHigherOrderComponent(
+	( Wrapped ) => ( props ) => {
+		const { siteId, user } = props;
+		const {
+			data: externalContributors = [],
+			isLoading,
+			isFetching,
+		} = useExternalContributorsQuery( siteId, { staleTime: 0 } );
+		const isUpdatingContributorStatus = isLoading || isFetching;
+		const { addExternalContributor } = useAddExternalContributorMutation();
+		const { removeExternalContributor } = useRemoveExternalContributorMutation();
+		const userId = user.linked_user_ID || user.ID;
+		const isExternalContributor =
+			userId && ! isUpdatingContributorStatus
+				? externalContributors?.includes( userId )
+				: undefined;
```

Comment:
There's a potential mismatch between query cache and API response which is why I pass in `undefined` and use `getDerivedStateFromProps` to correctly retrieve the actual value. This also fixes a bug which is currently present in prod where the component doesn't update after it retrieves the value from the API (it'd change from `false`, the default, to `true`) but doesn't update local state.